app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -695,9 +691,7 @@ class HomeFragment : Fragment(), CoroutineScope, AccountObserver {
     private fun scrollToCollections(addedTabsSize: Int) {
         launch {
             delay(ANIM_SCROLL_DELAY)
-            val tabsSize = requireComponents.core.sessionManager.sessions.filter {
-                (activity as HomeActivity).browsingModeManager.isPrivate == it.private
-            }.size
+            val tabsSize = getListOfTabs().size
```

Comment:
This call is a space & time complexity of `O(n)` (we create a new `Tab` object for each tab, only to throw them all away).

Suggest:

```kotlin
val tabSize = sessionManager.sessions.size
```

which may be still be `O(n)` (I haven't looked at the implementation of `sessions`), but you won't be creating any more objects.